### PR TITLE
KYLIN-4212 add user profile page allowing to change password

### PR DESCRIPTION
**Pull request by xiacongling**
see: https://issues.apache.org/jira/browse/KYLIN-4212

### CODE REVIEW

File: webapp/app/routes.json [hunk 1]
```diff
@@ -141,5 +141,13 @@
       "tab": "admin",
       "controller": "StreamingReceiverCtrl"
     }
+  },
+  {
+    "url": "/user/me",
```
Comment (lshmouse) [inline webapp/app/routes.json]: me -> profile

File: webapp/app/partials/header.html [hunk 2]
```diff
@@ -76,6 +76,7 @@
             </a>
             <ul class="user-menu dropdown-menu-right dropdown-menu dropdown-yellow dropdown-caret dropdown-close"
                 role="menu">
+              <li><a href="user/me"><i class="ace-icon fa fa-user"></i> Profile</a></li>
```
Comment (lshmouse) [inline webapp/app/partials/header.html]: me -> profile

File: webapp/app/js/controllers/userProfile.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      repeatPassword: '',
+      newPassword: ''
+    }
+  };
+  $scope.changePwd = function () {
+    $scope.changePwdUser = initChangePwdUserWithCurrent();
+    $scope.currentPwdNeeded = true;
+    $modal.open({
+      templateUrl: 'changePwd.html',
+      controller: changePwdCtrl,
```
Comment (lshmouse) [inline webapp/app/js/controllers/userProfile.js]: Should we logout after changing current password?

**Comment by xiacongling** [comment]
> Please upload a screen shot about the new profile. Thanks~

uploaded. pls view them on the issue page: https://issues.apache.org/jira/browse/KYLIN-4212

**Comment by xiacongling** [comment]
Password changing by user is restricted in 'testing' or 'custom' profile now. "BadRequestException: Action not allowed" would be raised in other profiles in previous commit.

**Comment by lshmouse** [comment]
@nichunen Could you help to review this PR? Thanks~

**Comment by nichunen** [comment]
@lshmouse OK

**Comment by nichunen** [comment]
屏幕快照 2019-12-15 下午5 19 00
@xiacongling Hi, I'm testing and review your pr, and it seems it does not work fine on my side.

**Comment by xiacongling** [comment]
> 屏幕快照 2019-12-15 下午5 19 00
> @xiacongling Hi, I'm testing and review your pr, and it seems it does not work fine on my side.

hi, @nichunen. It seems that `UserProfileCtrl` in `userProfile.js` is not registered. Could you try to clear the browser's cache or rebuild the frontend code? It works fine for me and looks like:
screenshot

**Comment by nichunen** [comment]
@xiacongling Still not work. Is it due to ng-controller not set for the profile page?

**Comment by xiacongling** [comment]
@nichunen OK. I try to add controller attribute in the html template and seems good, too. I'm not sure it is the reason for the error, since controller was set in `routes.json` in the previous commit. In the error message, the name of the controller `UserProfileCtrl` is shown, indicating ng knows what controller the template will use, but cannot find the 'instance' for it.

**Comment by xiacongling** [comment]
@nichunen Thanks a lot for the fix 👍

**Pull request merged**